buildtools/composer/composer.go

```diff
@@ -22,14 +27,52 @@ type Package struct {
 	Description string
 }
 
+// A Show structure has a list of dependencies reported by Composer.
 type Show struct {
-	Installed []Package
+	Installed []Package `json:"installed"`
 }
 
-//go:generate bash -c "genny -in=$GOPATH/src/github.com/fossas/fossa-cli/graph/readtree.go gen 'Generic=Package' | sed -e 's/package graph/package composer/' > readtree_generated.go"
+// Cmd implements the Composer interface.
+type Cmd struct {
+	Runner
+}
 
-func (c *Composer) Dependencies(dir string) ([]Package, map[Package][]Package, error) {
-	// Run `composer show --format=json --no-ansi` to get resolved versions
+func New(composerBinary string) Composer {
+	return Cmd{
+		Runner: realRunner(composerBinary),
+	}
+}
+
+// A Runner runs composer and returns the output.
+type Runner interface {
+	// RunShow runs the `show` command in the dir specified with optional additional arguments.
+	RunShow(dir string, args ...string) (stdout string, stderr string, err error)
```

Comment:
Your level of commenting is fantastic, but this one is not neccessarily true. Because `RunShow` here is a method in the `Runner` interface we do not know exactly how it will work under the surface. This would be better suited for line 58 where we are actually running the command.

Reply:
good catch, will clarify